lang/cs/Org.Apache.REEF.Wake/Remote/Impl/TcpClientConnectionException.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+
+namespace Org.Apache.REEF.Wake.Remote.Impl
+{
+    /// <summary>
+    /// Exception class for the case when Tcp Client connection fails 
+    /// after trying for some finite number of times.
+    /// For example, see TcpClientConnectionFactory.cs
+    /// </summary>
+    public sealed class TcpClientConnectionException : Exception
```

Comment:
> It s already passed to message. I checked and ToString() also writes the message. We are ok.

Yes, but this requires discipline at the call site. Which we sometimes lack :-)

Can we instead add to the user supplied message, e.g. `userMessage + "Tried {n} times"`?